### PR TITLE
dns-scout 5.8 (new formula)

### PR DESCRIPTION
**Pull request by careyjames**
- [x ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/homebrew-dns-scout.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class HomebrewDnsScout < Formula
```
Comment (carlocab) [inline Formula/homebrew-dns-scout.rb:1]:
```suggestion
class DnsScout < Formula
```
Just call this `dns-scout`. You'll need to change the filename too.

File: Formula/homebrew-dns-scout.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  desc "DNS Scout for Linux/MacOS pulls and displays DNS records"
+  homepage "https://github.com/careyjames/DNS-Scout"
+  url "https://github.com/careyjames/DNS-Scout/releases/download/v5.8/DNS-Scout-v5.8.tar.gz"
+  sha256 "e8d92f653069562699be58ca4fce421c645e7389aa7603fe786de046fbd3dc60"
+  license "MIT"
+  depends_on "go"
```
Comment (carlocab) [inline Formula/homebrew-dns-scout.rb:7]:
```suggestion
  license "MIT"

  depends_on "go"
```

File: Formula/homebrew-dns-scout.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  def install
+    ENV["GOOS"] = "darwin"
+    ENV["GOARCH"] = "amd64"
+    system "go", "build", "-o", "#{bin}/dns-scout"
+    chmod 0555, "#{bin}/dns-scout"
```
Comment (carlocab) [inline Formula/homebrew-dns-scout.rb:13]:
```suggestion
    system "go", "build", *std_go_args
```

File: Formula/homebrew-dns-scout.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  test do
+    system "#{bin}/dns-scout", "--version"
```
Comment (carlocab) [inline Formula/homebrew-dns-scout.rb:17]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

File: Formula/dns-scout.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+class HomebrewDnsScout < Formula
```
Comment (SMillerDev) [inline Formula/dns-scout.rb:1]:
```suggestion
class DnsScout < Formula
```

File: Formula/dns-scout.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  homepage "https://github.com/careyjames/DNS-Scout"
+  url "https://github.com/careyjames/DNS-Scout/releases/download/v5.8/DNS-Scout-v5.8.tar.gz"
+  sha256 "e8d92f653069562699be58ca4fce421c645e7389aa7603fe786de046fbd3dc60"
+  license "MIT"
+
+  depends_on "go"
+
+  def install
+    system "go", "build", *std_go_args
+    chmod 0555, "#{bin}/dns-scout"
```
Comment (SMillerDev) [inline Formula/dns-scout.rb:12]: Why?

File: Formula/dns-scout.rb [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  test do
+    foo build-foo input.foo
```
Comment (SMillerDev) [inline Formula/dns-scout.rb:16]: ???

File: Formula/dns-scout.rb (hunk 5 above)
Comment (careyjames) [reply Formula/dns-scout.rb]: ha ha ooops
sorry guys, im a virgin! my first

File: Formula/dns-scout.rb (hunk 6 above)
Comment (careyjames) [reply Formula/dns-scout.rb:12]: it had errors about an executable ?

File: Formula/dns-scout.rb [hunk 8]
```diff
@@ -0,0 +1,18 @@
+class DnsScout < Formula
+  desc "DNS Scout for Linux/MacOS pulls and displays DNS records"
```
Comment (carlocab) [inline Formula/dns-scout.rb:2]:
```suggestion
  desc "Pulls and display DNS records"
```

**Comment by careyjames** [comment]
im sorry i didnt realize i could tap commit suggestions..

**Comment by p-linnane** [comment]
I've moved the formula to the `d/` directory, fixed the build, and squashed the commits. You will need to figure out a suitable test since arguments can't be passed to the application directly.

**Comment by careyjames** [comment]
ok is this ok: for mac os intel and silicon?

class DnsScout < Formula
  desc "Pulls and display DNS records"
  homepage "https://github.com/careyjames/DNS-Scout"
  url "https://github.com/careyjames/DNS-Scout/archive/refs/tags/v5.8.tar.gz"
  sha256 "f8e2c2f68c2e402b8aeef0225ebf04aedb6c6852b102e2c7f3dbe6f6f1e60ec9"
  sha256 "45b5fe937a570319da3b2957681010adcf4b077cc8bad59e5a1f445e18a951ab"
  license "MIT"

  depends_on "go" => :build

  def install
    system "go", "build", *std_go_args(ldflags: "-s -w")
  end
end

**Comment by careyjames** [comment]
actually this?

if Hardware::CPU.arm?
    url "https://github.com/careyjames/dns-scout/releases/download/v5.8/dns-scout-macos-arm64-v5.8.tar.gz"
    sha256 "your_arm64_sha256_here"
  else
    url "https://github.com/careyjames/dns-scout/releases/download/v5.8/dns-scout-macos-amd64-v5.8.tar.gz"
    sha256 "your_amd64_sha256_here"
  end

**Comment by SMillerDev** [comment]
> We frown on authors submitting their own work unless it is very popular.


https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff

**Pull request closed without merging**